src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs

```diff
@@ -733,10 +755,45 @@ internal async Task SearchPackagesAndRefreshUpdateCountAsync(string searchText,
                 loadContext.CachedPackages = Model.CachedUpdates;
             }
 
-            var packageFeed = await CreatePackageFeedAsync(loadContext, _topPanel.Filter, _uiLogger);
+            // only make recommendations when
+            //   the single source repository is nuget.org,
+            //   the package manager was opened for a project, not a solution,
+            //   this is the Browse tab,
+            //   and the search text is an empty string
+            _recommendPackages = false;
+            if (loadContext.IsSolution == false
+                && _topPanel.Filter == ItemFilter.All
+                && searchText == string.Empty
+                && loadContext.SourceRepositories.Count() == 1
+                && TelemetryUtility.IsNuGetOrg(loadContext.SourceRepositories.First().PackageSource))
+            {
+                // also check if this is a PC-style project. We will not provide recommendations for PR-style
+                // projects until we have a way to get dependent packages without negatively impacting perf.
+                var Projects = (Model.Context.Projects ?? Enumerable.Empty<NuGetProject>()).ToArray();
+                foreach(var project in Projects)
+                {
+                    if (project.ProjectStyle == ProjectModel.ProjectStyle.PackagesConfig)
+                    {
+                        _recommendPackages = true;
+                    }
+                }
+            }
+
+            // Check for A/B experiment here. For control group, call CreatePackageFeedAsync with false instead of _recommendPackages
+            // Also check if the environment variable RecommendNuGetPackages is set to 1
+            var packageFeeds = (mainFeed: (IPackageFeed)null, recommenderFeed: (IPackageFeed)null);
+            if (IsRecommenderFlightEnabled() || _forceRecommender)
```

Comment:
Maybe just capture it all in one method?

Reply:
I moved the _forceRecommender check inside IsRecommenderFlightEnabled(). Is that what you meant?

Reply:
Yes.